HelpSource/Reference/ServerPluginAPI.schelp

```diff
@@ -0,0 +1,189 @@
+title:: Server Plugin API
+summary:: Writing unit generators
+categories:: Internals
+
+section:: Input rates
+
+These four constants identify the calculation rates of inputs in SuperCollider.
+
+definitionlist::
+## code::calc_ScalarRate::
+|| Initial rate. Conventionally known in the language as ".ir".
+
+## code::calc_BufRate::
+|| Control rate. Conventionally known in the language as ".kr". 
+
+## code::calc_FullRate::
+|| Audio rate. Conventionally known in the language as ".ar".
+
+## code::calc_DemandRate::
+|| Demand rate. 
+::
+
+section:: UGen I/O
+
+definitionlist::
+## code::IN(index)::
+|| A single block of audio-rate input as a float* at the given index. Index 0 is the first input to the ugen, index 1 the second input, and so forth.
+
+## code::IN0(index)::
+|| A single sample of control-rate input as a float, at the given index.
+
+## code::OUT(index)::
+|| A single block of audio-rate output as a float* at the given index.
+
+## code::OUT0(index)::
+|| A single sample of control-rate input as a float, at the given index.
+
+## code::INRATE(index)::
+|| Get the rate of a given input index. This will be one of the four rates.
+::
+
+section:: Unary operators
```

Comment:
stl: https://en.wikipedia.org/wiki/Standard_Template_Library

e.g. there is `std::nan`: http://en.cppreference.com/w/cpp/numeric/math/nan